sdlib/d/gc/tstate.d

```diff
@@ -255,6 +255,13 @@ unittest suspend {
 
 	auto autoResumeThreadID = runThread(autoResume);
 
+	scope(exit) {
+		mustStop.store(1);
+
+		void* ret;
+		pthread_join(autoResumeThreadID, &ret);
+	}
+
```

Comment:
Note, I found while running unittests, this had a segfault when a test fails, because the thread continues to look at the now-missing stack data for `mustStop`. So I rewrote it this way so a failure doesn't segfault.

Debugging segfaults on sdunit is so bad, because a debugger has no idea what's going on. All stack frames are ??

Reply:
#400